doc/sphinx/addendum/nsatz.rst

```diff
@@ -32,6 +32,36 @@ Nsatz: tactics for proving equalities in integral domains
 
    doing automatic introductions.
 
+   `radicalmax`
+     bound when searching for r such that
+     :math:`c (P−Q) r = \sum_{i=1..s} S_i (P i − Q i)`
+
+   `strategy`
+     gives the order on variables :math:`X_1,\ldots,X_n` and the strategy
+     used in Buchberger algorithm (see :cite:`sugar` for details):
+
+       * strategy = 0: reverse lexicographic order and newest s-polynomial.
+       * strategy = 1: reverse lexicographic order and sugar strategy.
+       * strategy = 2: pure lexicographic order and newest s-polynomial.
+       * strategy = 3: pure lexicographic order and sugar strategy.
+
+   `parameters`
+     list of variables :math:`X_{i_1},\ldots,X_{i_k}` among
+     :math:`X_1,\ldots,X_n` which are considered as parameters: computation will be performed with
+     rational fractions in these variables, i.e. polynomials are considered
+     with coefficients in :math:`R(X_{i_1},\ldots,X_{i_k})`. In this case, the coefficient
+     :math:`c` can be a non constant polynomial in :math:`X_{i_1},\ldots,X_{i_k}`, and the tactic
+     produces a goal which states that :math:`c` is not zero.
+
+   `variables`
+     list of the variables in the decreasing order in
+     which they will be used in the Buchberger algorithm. If `variables` = :g:`(@nil R)`,
+     then `lvar` is replaced by all the variables which are not in
+     `parameters`.
```

Comment:
This should be made clearer that `parameters` and `variables` must be lists in the sense that they must be of type `@list R`. For now, this is only hinted at with the case of `variables := (@nil R)`.

Reply:
This actually uses `one_term`, not `term`.  But it's still strange in that the term `(A B C)` means apply a function A to B and C (right?), while nsatz looks like it's just using that as a list.  Also, I don't get what the :: means here (taken from the Nsatz.v test, which is used in all 4 examples in that file that explicitly specify all the parameters:

```
Time nsatz with radicalmax :=1%N strategy:=1%Z
  parameters:=(X O::X B::X C::nil)
  variables:= (@nil R).
```

Reply:
What does the `@list` or `@nil` construction mean?  Function application?  Disabling implicit arguments?

Reply:
The examples are not using the list notations, that's why they are confusing. Yes, `@` is for disabling implicit notations. So `@nil R` is the empty list of real numbers. And `a :: nil` is the list containing only `a`.

> But it's still strange in that the term `(A B C)` means apply a function A to B and C (right?), while nsatz looks like it's just using that as a list.

I don't think that's the case: it seems to me that nsatz requires objects of type `list R` as arguments.

Reply:
It is indeed a bit exposing implementation details which don't necessarily have to be exposed. Why not to let the tactic notation build the list itself, using e.g. Ltac2 notations, or making a slight extension of `Ltac Notation`. Cc @ppedrot.

Reply:
> This should be made clearer that `parameters` and `variables` must be lists in the sense that they must be of type `@list R`.

I can add language like this, but there's still a disconnect because `[X B]` doesn't match the wording.

```a list of lists, each of type `@list R`, containing the variables in the decreasing order in```

> The examples are not using the list notations

I will fix that for the next round.

Reply:
It was a nice idea to use normal list notation like `[X B]` but it doesn't parse that way, perhaps because `nsatz` is defined with `Tactic Notation` (the error is on the second line shown, at the "O".

```
Time nsatz with radicalmax :=1%N strategy:=1%Z
  parameters:=(X [O; X] [B; X] [C])
  variables:= (@[] R).

File "./success/Nsatz.v", line 297, characters 19-20:
Error:
Syntax error: ']' expected after [constr:operconstr level 200] (in [constr:operconstr]).
```

Reply:
List notations are not available by default. You need `Require List. Import List.ListNotations.`

Reply:
Thanks, but it still doesn't work.  Error on `[O; X]`:

```
Time nsatz with radicalmax :=1%N strategy:=1%Z
  parameters:=(X [O; X] [B; X] [C])
  variables:= (@nil    R).

File "./success/Nsatz.v", line 299, characters 17-23:
Error:
The term "[X]" has type "list (point -> R)" while it is expected to have type
 "list point".
```

Reply:
I think you got confused by the strange text or the strange examples, but this is a simple list of reals that must be passed, and spaces do mean function application and nothing else. So your parameters value is indeed not correct.